### PR TITLE
Fix bug in SSIM

### LINKED ISSUE #203: Split code for SSIM and MS-SSIM into separate files
**Is your feature request related to a problem? Please describe.**
Right now `piq/ssim.py` is the biggest file in our library with ~600 lines of code. Tests for this file in `tests/test_ssim` also span for ~500 lines of code. We initially combined SSIM and MS-SSIM because they were simular to each other, but now with release of IW-SSIM and SR-SIM metrics it's seems logical to split them. 

**Describe the solution you'd like**
Move all MS-SSIM code into `piq/ms_ssim.py` and all tests into `tests/test_ms_ssim.py`.

**Additional context**
We can do this after all current PR's are merged (#184 #195 #201 #202 ) to avoid confusion.

### LINKED ISSUE #209: Bug in SSIM: Can't reproduce paper values
**Describe the bug**
Performance of SSIM on TID2013 dataset:
PIQ - SRCC **0.5544**  KRCC **0.3883**
Original paper - SRCC **0.637** KRCC **0.464**
DISTS paper - SRCC **0.727** KRCC **0.545**

**To Reproduce**
Code is not available. Will be released soon (see #207).

**Expected behavior**
Match performance to original paper

### PR DESCRIPTION
**Pull request by zakajd**
Closes #209 and closes #203 

## Proposed Changes

  - Split code for SSIM and MS-SSIM
  - Fix bug (which ocured to to missing average pooling. See https://github.com/scikit-image/scikit-image/issues/5192 for details
  - Rename variables to have a more meaningful names

TODO:
- [x] Add tests to compare with MATLAB
- [ ] Update table in README

### CODE REVIEW

**Comment by zakajd** [comment]
Ready for review.
Test failed due to duplicate code in `tests/test_ssim.py` and `tests/test_ms_ssim.py`. This will be fixed during test refactoring (#97 )

File: piq/ssim.py [hunk 1]
```diff
@@ -382,69 +208,26 @@ def _ssim_per_channel(x: torch.Tensor, y: torch.Tensor, kernel: torch.Tensor,
...
-    # weights, (level)
-    msssim_val = torch.prod((mcs_ssim ** scale_weights_tensor.view(-1, 1, 1)), dim=0).mean(1)
-
-    return msssim_val
+    ssim_val = ss.mean(dim=(-1, -2))
+    cs = cs.mean(dim=(-1, -2))
+    return ssim_val, cs
 
 
 def _ssim_per_channel_complex(x: torch.Tensor, y: torch.Tensor, kernel: torch.Tensor,
```
Comment (denproc) [inline piq/ssim.py:233]: Shouldn't we update the complex-valued SSIM similarly to SSIM?

File: piq/ssim.py [hunk 2]
```diff
@@ -150,213 +161,28 @@ def __init__(self, kernel_size: int = 11, kernel_sigma: float = 1.5, k1: float =
...
-
-        score = multi_scale_ssim(x=prediction, y=target, kernel_size=self.kernel_size, kernel_sigma=self.kernel_sigma,
-                                 data_range=self.data_range, reduction=self.reduction, scale_weights=self.scale_weights,
-                                 k1=self.k1, k2=self.k2)
-        return torch.ones_like(score) - score
-
-
 def _ssim_per_channel(x: torch.Tensor, y: torch.Tensor, kernel: torch.Tensor,
                       data_range: Union[float, int] = 1., k1: float = 0.01,
                       k2: float = 0.03) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
```
Comment (denproc) [inline piq/ssim.py:188]: In case of avoiding Unions as much as possible, we should replace it with `List[torch.Tensor]`.

**Review by denproc — CHANGES_REQUESTED** [review]
Nice catch with downsampling bug!
Minor comments about the documentation is to be resolved. Others are optional and can be discussed.

**Comment by snk4tr** [comment]
Hey guys. Sonar check fails because of code duplication. As far as I can see, all duplications are in tests, which is absolutely fine with me. Let me know if you think different. Otherwise I will modify Sonar rules to ignore duplications in tests

**Comment by zakajd** [comment]
@snk4tr  disagree.
Let's leave Sonar cloud checks for feature extractors and add proper testing for models instead. At least check that forward call runs without failure.
We can also move new models there (like PieAPP)

**Comment by snk4tr** [comment]
@denproc what do you think?

**Comment by snk4tr** [comment]
> @denproc what do you think?

Ok, I see that @denproc liked the idea of not removing feature extractors from the coverage report. Reverted this change.

**Comment by snk4tr** [comment]
@zakajd are you working on the one? Let's finilize this one. Let me know if you need any help.

**Comment by zakajd** [comment]
@snk4tr this one can be merged if there is no failing tests. 

I didn't renamed variables in ms-ssim as proposed by Denis because its better to do that in a separate PR (along with some other renamings).

**Comment by snk4tr** [comment]
@zakajd are you still working on this one or it is ready?

**Comment by zakajd** [comment]
@snk4tr  it's ready

**Review by snk4tr — APPROVED** [review]
The PR looks good, ready to merge.
@zakajd FYI: it is easier to follow PRs when they are make to solve a single issue. In this case, the PR is both about fixing the bug and refactoring, which confuses things a little. Consider making separate PRs for these two next time. 
P.S. If you did that because of long review time, do not let long processes to break your style 😉

**Pull request merged**